src/jaxsim/parsers/descriptions/link.py

```diff
@@ -31,7 +31,7 @@ class LinkDescription(JaxsimDataclass):
     mass: float = dataclasses.field(repr=False)
     inertia: jtp.Matrix = dataclasses.field(repr=False)
     index: int | None = None
-    parent: LinkDescription | None = dataclasses.field(default=None, repr=False)
+    parent_name: str | None = dataclasses.field(default=None, repr=False)
```

Comment:
In #351 the `jax_dataclasses` will be removed, but for the time being we should mark the string attribute as static, since it's not a JAX type. This is not a problem for the time being since we don't have any JIT-ted method anyway
```suggestion
    parent_name: Static[str | None] = dataclasses.field(default=None, repr=False)
```